components/brave_shields/browser/https_everywhere_service.cc

```diff
@@ -12,83 +12,70 @@
 #include "base/threading/scoped_blocking_call.h"
-#include "base/values.h"
-#include "third_party/leveldatabase/src/include/leveldb/db.h"
-#include "third_party/re2/src/re2/re2.h"
+#include "brave/browser/component_updater/brave_component_installer.h"
+#include "brave/common/pref_names.h"
+#include "brave/vendor/https-everywhere-lib-cpp/src/wrapper.hpp"
+#include "chrome/browser/browser_process_impl.h"
+#include "chrome/browser/first_run/first_run.h"
+#include "components/component_updater/component_updater_service.h"
+#include "components/prefs/pref_service.h"
 #include "third_party/zlib/google/zip.h"
 
-#define DAT_FILE "httpse.leveldb.zip"
-#define DAT_FILE_VERSION "6.0"
+#define DAT_FILE "httpse-rs.json.zip"
+#define DAT_FILE_VERSION "7.0"
 #define HTTPSE_URLS_REDIRECTS_COUNT_QUEUE   1
 #define HTTPSE_URL_MAX_REDIRECTS_COUNT      5
 
 namespace {
 
-std::vector<std::string> Split(const std::string& s, char delim) {
-  std::stringstream ss(s);
-  std::string item;
-  std::vector<std::string> result;
-  while (getline(ss, item, delim)) {
-    result.push_back(item);
-  }
-  return result;
-}
+// Legacy component ID, unregistered at startup
+const char kHTTPSEverywhere6ComponentBase64PublicKey[] =
+    "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAvn9zSMjTmhkQyrZu5UdN"
+    "350nPqLoSeCYngcC7yDFwaUHjoBQXCZqGeDC69ciCQ2mlRhcV2nxXqlUDkiC6+7m"
+    "651nI+gi4oVqHagc7EFUyGA0yuIk7qIMvCBdH7wbET27de0rzbRzRht9EKzEjIhC"
+    "BtoPnmyrO/8qPrH4XR4cPfnFPuJssBBxC1B35H7rh0Br9qePhPDDe9OjyqYxPuio"
+    "+YcC9obL4g5krVrfrlKLfFNpIewUcJyBpSlCgfxEyEhgDkK9cILTMUi5vC7GxS3P"
+    "OtZqgfRg8Da4i+NwmjQqrz0JFtPMMSyUnmeMj+mSOL4xZVWr8fU2/GOCXs9gczDp"
+    "JwIDAQAB";
 
-// returns parts in reverse order, makes list of lookup domains like com.foo.*
-std::vector<std::string> ExpandDomainForLookup(const std::string& domain) {
-  std::vector<std::string> resultDomains;
-  std::vector<std::string> domainParts = Split(domain, '.');
-  if (domainParts.empty()) {
-    return resultDomains;
-  }
+void NoopReadyCallback(const base::FilePath&, const std::string&) {}
 
-  for (size_t i = 0; i < domainParts.size() - 1; i++) {
-    // i < size()-1 is correct: don't want 'com.*' added to resultDomains
-    std::string slice = "";
-    std::string dot = "";
-    for (int j = domainParts.size() - 1; j >= static_cast<int>(i); j--) {
-      slice += dot + domainParts[j];
-      dot = ".";
-    }
-    if (0 != i) {
-      // We don't want * on the top URL
-      resultDomains.push_back(slice + ".*");
-    } else {
-      resultDomains.push_back(slice);
-    }
-  }
-  return resultDomains;
+void OnLegacyHTTPSEverywhereRegistered(
+    scoped_refptr<component_updater::ComponentInstaller> installer
+) {
+  installer->Uninstall();
 }
-std::string leveldbGet(leveldb::DB* db, const std::string &key) {
-  if (!db) {
-    return "";
-  }
 
-  std::string value;
-  leveldb::Status s = db->Get(leveldb::ReadOptions(), key, &value);
-  return s.ok() ? value : "";
+void MigrateHTTPSEverywhere6Prefs() {
```

Comment:
the service shouldn't be responsible for the component, this code should go in `BraveComponentLoader`